src/amo/components/Addon/index.js

```diff
@@ -60,6 +60,14 @@ import ShowMoreCard from 'ui/components/ShowMoreCard';
 import './styles.scss';
 
 
+// Find out if slug is invalid.
+const isInvalidSlugID = (slug) => {
+  // eslint-disable-next-line no-restricted-globals
+  return slug && !isNaN(slug)
+    && ((typeof slug === 'string' && slug.charAt(0) !== '-')
+    || (typeof slug === 'number'));
```

Comment:
We should have a test case for this too and I don't see one.

Reply:
thanks i'll take another look..

Reply:
hey @willdurand  - do you mean if it's actually a number? I guess i thought this test was covered under `dispatches a server redirect when slug is a numeric ID` - or let me know which piece.. maybe i misunderstood..?

Reply:
it also checks for all these pieces with the following tests:
`dispatches a server redirect when slug is a numeric ID`
`dispatches a server redirect when slug is a stringified integer`
`does not dispatch a server redirect when slug is a stringified integer that starts with -`

 i believe this covers all these pieces? are you saying these should be combined into one test?

Reply:
It is probably fine. I was a bit confused because we did not check the type before but you have to do it because you check the first char.

Reply:
Maybe we could get rid of the parentheses btw

Reply:
right..i added type check b/c of charAt piece...

i actually had some less parentheses but linting complained (`Unexpected mix of '&&' and '||'  no-mixed-operators`) unless i did i this way..